vividus-plugin-saucelabs/src/main/java/org/vividus/selenium/sauce/SauceLabsCapabilitiesConfigurer.java

```diff
@@ -0,0 +1,123 @@
+import org.vividus.selenium.event.WebDriverQuitEvent;
+import org.vividus.ui.web.configuration.WebApplicationConfiguration;
+
+public class SauceLabsCapabilitiesConfigurer implements DesiredCapabilitiesConfigurer
+{
+    private static final String SAUCE_OPTIONS = "sauce:options";
+    private static final int DEFAULT_HTTPS_PORT = 443;
+
+    private final WebApplicationConfiguration webApplicationConfiguration;
+    private final IBddRunContext bddRunContext;
+    private final SauceConnectManager sauceConnectManager;
+    private boolean sauceLabsEnabled;
+    private boolean sauceConnectEnabled;
+    private String restUrl;
+
+    public SauceLabsCapabilitiesConfigurer(WebApplicationConfiguration webApplicationConfiguration,
+            IBddRunContext bddRunContext, SauceConnectManager sauceConnectManager)
+    {
+        this.webApplicationConfiguration = webApplicationConfiguration;
+        this.bddRunContext = bddRunContext;
+        this.sauceConnectManager = sauceConnectManager;
+    }
+
+    @Subscribe
+    public void stopSauceConnect(WebDriverQuitEvent event)
+    {
+        sauceConnectManager.stop();
+    }
+
+    @Override
+    public void addCapabilities(DesiredCapabilities desiredCapabilities)
+    {
+        if (sauceLabsEnabled)
+        {
+            Proxy proxy = (Proxy) desiredCapabilities.getCapability(CapabilityType.PROXY);
+            if (sauceConnectEnabled || proxy != null)
+            {
+                SauceConnectOptions options = createSauceConnectOptions(proxy);
+                sauceConnectManager.start(options);
+                addSauceOption(desiredCapabilities, "tunnelIdentifier", sauceConnectManager.getTunnelId());
+                desiredCapabilities.setCapability(CapabilityType.PROXY, (Object) null);
+            }
+            RunningStory runningStory = bddRunContext.getRunningStory();
+            if (runningStory != null)
+            {
+                addSauceOption(desiredCapabilities, "name", runningStory.getName());
+            }
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private void addSauceOption(DesiredCapabilities desiredCapabilities, String capabilityName, Object value)
+    {
+        Map<String, Object> sauceOptions = (Map<String, Object>) desiredCapabilities.getCapability(SAUCE_OPTIONS);
+        if (sauceOptions == null)
+        {
+            sauceOptions = new HashMap<>();
+            desiredCapabilities.setCapability(SAUCE_OPTIONS, sauceOptions);
+        }
+        sauceOptions.put(capabilityName, value);
+    }
+
+    private SauceConnectOptions createSauceConnectOptions(Proxy proxy)
+    {
+        SauceConnectOptions sauceConnectOptions = new SauceConnectOptions();
+        sauceConnectOptions.setHost(webApplicationConfiguration.getHost());
+        if ("https".equals(webApplicationConfiguration.getMainApplicationPageUrl().getScheme()))
+        {
+            sauceConnectOptions.setPort(DEFAULT_HTTPS_PORT);
```

Comment:
can't we get rid of http and use https only since many modern web sites refuse usage of not secure protorol?